core/vtk/ttkMergeTreeClustering/ttkMergeTreeClustering.cpp

```diff
@@ -0,0 +1,707 @@
+  std::vector<vtkMultiBlockDataSet *> &inputTrees2) {
+  if(not isDataVisualizationFilled())
+    runCompute<dataType>(outputVector, inputTrees, inputTrees2);
+  runOutput<dataType>(outputVector, inputTrees, inputTrees2);
+  return 1;
+}
+
+template <class dataType>
+int ttkMergeTreeClustering::runCompute(
+  vtkInformationVector *outputVector,
+  std::vector<vtkMultiBlockDataSet *> &inputTrees,
+  std::vector<vtkMultiBlockDataSet *> &inputTrees2) {
+  int verbose = 0;
+
+  // ------------------------------------------------------------------------------------
+  // --- Construct trees
+  // ------------------------------------------------------------------------------------
+  if(verbose > 0)
+    std::cout << "Construct trees" << std::endl;
+
+  const int numInputs = inputTrees.size();
+  const int numInputs2 = inputTrees2.size();
+
+  setDataVisualization(numInputs, numInputs2);
+
+  std::vector<MergeTree<dataType>> intermediateMTrees(numInputs),
+    intermediateMTrees2(numInputs2), barycenters(NumberOfBarycenters);
+  std::vector<FTMTree_MT *> intermediateTrees(numInputs),
+    intermediateTrees2(numInputs2);
+
+  constructTrees<dataType>(
+    inputTrees, intermediateMTrees, treesNodes, treesArcs, treesSegmentation);
+  constructTrees<dataType>(inputTrees2, intermediateMTrees2, treesNodes2,
+                           treesArcs2, treesSegmentation2);
+
+  intermediateTrees = mergeTreeToFTMTree<dataType>(intermediateMTrees);
+  intermediateTrees2 = mergeTreeToFTMTree<dataType>(intermediateMTrees2);
+
+  // ------------------------------------------------------------------------------------
+  // --- Call base
+  // ------------------------------------------------------------------------------------
+  if(verbose > 0)
+    printMsg("Call base");
+  bool AddNodes = true;
+  // Classical distance
+  double distance = 0;
+
+  // Verify parameters
+  if(Backend == 0) {
+    BranchDecomposition = true;
+    NormalizedWasserstein = true;
+    KeepSubtree = false;
+  } else if(Backend == 1) {
+    BranchDecomposition = false;
+    NormalizedWasserstein = false;
+    KeepSubtree = true;
+    ComputeBarycenter = false;
+  }
+  if(ComputeBarycenter) {
+    if(not BranchDecomposition)
+      printMsg("BranchDecomposition is set to true since the barycenter "
+               "computation is asked.");
+    BranchDecomposition = true;
+    if(KeepSubtree)
+      printMsg("KeepSubtree is set to false since the barycenter computation "
+               "is asked.");
+    KeepSubtree = false;
+  }
+  if(not BranchDecomposition) {
+    if(NormalizedWasserstein)
+      printMsg("NormalizedWasserstein is set to false since branch "
+               "decomposition is not asked.");
+    NormalizedWasserstein = false;
+  }
+  EpsilonTree2 = EpsilonTree1;
+  Epsilon2Tree2 = Epsilon2Tree1;
+  Epsilon3Tree2 = Epsilon3Tree1;
+
+  // Call base
+  if(not ComputeBarycenter) {
+    MergeTreeDistance mergeTreeDistance;
+    mergeTreeDistance.setVerbose(Verbose);
```

Comment:
What is that function doing?
Have you seen the `setDebugLevel()` function (inherited from the base `Debug` class) which may have a similar motivation?

Reply:
The `verbose` of the mergeTreeClustering filter will be removed and replaced by the use of debugLevel_ thanks!